Example datasets page revamp (#414)
neo4j/docs-getting-started: modules/ROOT/pages/appendix/example-data.adoc

[[example-data]]
= Example datasets
:description: See a list of available example datasets for Neo4j and learn how to import and explore them.
:page-pagination: previous

//Check Mark
:check-mark: icon:check[]

//Cross Mark
:cross-mark: icon:times[]

Neo4j offers a variety of example datasets to get started with the products.
Some of them are built-in, others can be retrieved from a GitHub repository or accessed from the link:https://demo.neo4jlabs.com:7473[demo server].
This page shows how to access them.

== Available datasets

This list includes all available datasets, descriptions, and how to access them from different platforms.

[IMPORTANT]
====
Some datasets are not continuously maintained.

> Information might be outdated.

Reviewer: ```suggestion
Information may be outdated.
```

| A graph example using a dataset of movie reviews for generating personalized, real-time recommendations.

| link:https://github.com/neo4j-graph-examples/pole[Crime investigation (POLE)]
| {check-mark}
| {cross-mark}
| {cross-mark}
| A Persons Objects Locations Events example data model focused on the relationships between people, objects, locations and events.

s| Healthcare analysis
| {check-mark}
| {cross-mark}
| {cross-mark}
| An example graph using FDA Adverse Event Reporting System (FAERS) datasets.
It contains information on adverse event and medication error reports submitted to FDA.

| link:https://github.com/neo4j-examples/game-of-thrones[Game of Thrones]
| {cross-mark}
m| :guide got
a| link:https://demo.neo4jlabs.com:7473/browser/?dbms=neo4j://gameofthrones@demo.neo4jlabs.com&db=gameofthrones[`gameofthrones`]
| A graph based on George R. R. Martin's series Game of Thrones.
It contains information on the interaction between the characters throughout the books.

| link:https://github.com/jexp/fincen[FinCEN]
| {cross-mark}
| {cross-mark}
a| link:https://demo.neo4jlabs.com:7473/browser/?dbms=neo4j://fincen@demo.neo4jlabs.com&db=fincen[`fincen`]

> | A graph with data from the global investigation FinCEN Files for money laundering.

Reviewer: ```suggestion
| A graph with data from the global investigation FinCEN Files concerning money laundering.
```
else it sounds like a how-to on money laundering 😆

Author: oops 😂

| A graph using data from WordNet, a large lexical database of English. 
It groups nouns, verbs, adjectives, and adverbs into sets of cognitive synonyms (synsets), each expressing a distinct concept.

| link:https://github.com/neo4j-graph-examples/icij-paradise-papers[Panama Papers]
| {cross-mark}
| {cross-mark}
| {cross-mark}
| The Paradise Papers dataset and guide from the International Consortium of Investigative Journalists (ICIJ).

| link:https://github.com/neo4j-partners/neo4j-transport-for-london[London public transportation network]
| {cross-mark}
| {cross-mark}
| {cross-mark}
| A graph of the London public transportation network containing information on statios and tube lines.

| link:https://github.com/neo4j-graph-examples/network-management[IT management]
| {cross-mark}
| {cross-mark}
| {cross-mark}
| An example graph representing a network and IT management.
It contains information for dependency and root cause analysis, and more.

|===

== Built-in examples

Some example datasets can be retrieved directly from Workspace and Neo4j Browser in the form of guides.
With them, you can create or import a dataset and learn how to explore it.

=== Neo4j Workspace

> When running your link:{docs-home}/aura[Aura instance], use the academic cap icon on the top right section to access the interactive guides.

Reviewer: ```suggestion
From the Aura Console, Workspace can be accessed from your link:{docs-home}/aura[Aura instance] via the `Open` button.
Once in Workspace, use the academic cap icon on the top right to access the interactive guides.
```

> There you will find interactive guides that will help you create several of the xref:appendix/example-data.adoc#_available_datasets[available datasets] and learn how to work with them.

Reviewer: ```suggestion
There you find interactive guides that contain several of the xref:appendix/example-data.adoc#_available_datasets[available datasets] and learn how to work with them.
```

image:workspace-guides.png[]

=== Neo4j Browser

Use the command `:guide` to access a list of built-in interactive guides or access them directly using these previously listed xref:appendix/example-data.adoc#_available_datasets[commands].

image:browser-guides.png[]

== Demo server

Optionally, you can access the demo server on https://demo.neo4jlabs.com:7473 to explore a number of datasets with read-only access for public use.

The username and password are the same as the dataset name.
For instance, for the `recommendations` dataset the username is `recommendations` and password is `recommendations` too.

Find the full list of datasets and the username/password entries to use in the xref:appendix/example-data.adoc#_available_datasets[available datasets] table.

== Database dump files

> In the GitHub repository http://github.com/neo4j-graph-examples[Neo4j graph examples], you find dump files for several graph example datasets, including the ones listed previously in the xref:appendix/example-data.adoc#_available_datasets[available datasets] table.

Reviewer: ```suggestion
In the GitHub repository link:http://github.com/neo4j-graph-examples[Neo4j graph examples], you find dump files for several graph example datasets, including the ones listed previously in the xref:appendix/example-data.adoc#_available_datasets[available datasets] table.
```

There are several ways to load them, depending on the environment that is being used:

* link:{docs-home}/aura/aurads/importing-data/import-db/#_import_database[Aura]
* link:{docs-home}/desktop-manual/current/operations/create-from-dump/[Neo4j Desktop]
* link:{docs-home}/operations-manual/current/kubernetes/operations/dump-load/#kubernetes-neo4j-load[Kubernetes]
* link:{docs-home}/operations-manual/current/docker/dump-load/[Docker]
* link:{docs-home}/operations-manual/current/backup-restore/restore-dump/[Neo4j Admin]

You can also refer to the link:{docs-home}import/[Importing your data] section to learn more ways to load a dataset to your instance, including other supported file formats.

[IMPORTANT]
====
The Neo4j version of some of the datasets might be older than your Neo4j version.

> In this case, you might need to configure Neo4j to upgrade your database by using `neo4j-admin database migrate` command.

Reviewer: What does this mean? Do the datasets contain more than just the data?

Author: I'm not sure what it means. It was already in the page. Maybe @renetapopova knows something about this, as it's about upgrading Neo4j?

Reviewer 2: I think this note concerns cases when you load databases that are on a previous major version or a different store format. So, if you're loading a 4.4 database into a Neo4j 5.23, you must migrate it after loading it into Neo4j.

Reviewer: I had a look in the repo and they're not all dump files, you can download zipfiles that contain csv files as well. So it could be worthwhile to specify that this concern only applies to dump files?

Author: Yes, it's under the dump file section, but no harm in stressing this again in the note

Reviewer 2: Sorry, I have completely forgotten to look at it.

Note that this command is run only on a stopped database.

> For more details, see link:https://neo4j.com/docs/operations-manual/current/tools/neo4j-admin/migrate-database/[Operations manual -> Tools].

Reviewer 2: ```suggestion
For more details, see https://neo4j.com/docs/upgrade-migration-guide/current/version-5/migration/migrate-databases/[Upgrade and Migration guide -> Migrate your databases].
```

Reviewer 2: And btw, this is about self-managed deployments.

> The Neo4j version of some of the dump files might be older than your Neo4j version.

Reviewer: ```suggestion
The Neo4j version of some of the dump files may be older than your Neo4j version.
```

> In this case, you might need to configure Neo4j to upgrade your database by using `neo4j-admin database migrate` command.

Reviewer 2: ```suggestion
In this case, you might need to upgrade your database dumps by using `neo4j-admin database migrate` command before you load it into Neo4j.
```

> In this case, you might need to upgrade your database dump using the `neo4j-admin database migrate` command before loading it into Neo4j.

Reviewer: ```suggestion
In this case, you need to upgrade your database dump using the `neo4j-admin database migrate` command before loading it into Neo4j.
```

> Note that this command is run only on a stopped database.

Reviewer: ```suggestion
Note that this command can only be run on a stopped database.
```

There are several ways to load them, depending on the environment that is being used:

* link:{docs-home}/aura/aurads/importing-data/import-db/#_import_database[Aura]
* link:{docs-home}/desktop-manual/current/operations/create-from-dump/[Neo4j Desktop]
* link:{docs-home}/operations-manual/current/kubernetes/operations/dump-load/#kubernetes-neo4j-load[Kubernetes]
* link:{docs-home}/operations-manual/current/docker/dump-load/[Docker]
* link:{docs-home}/operations-manual/current/backup-restore/restore-dump/[Neo4j Admin]

You can also refer to the link:{docs-home}import/[Importing your data] section to learn more ways to load a dataset to your instance, including other supported file formats.

[IMPORTANT]
====
The Neo4j version of some of the dump files may be older than your Neo4j version.
In this case, you need to upgrade your database dump using the `neo4j-admin database migrate` command before loading it into Neo4j.
Note that this command can only be run on a stopped database.

> For more details, see https://neo4j.com/docs/upgrade-migration-guide/current/version-5/migration/migrate-databases/[Upgrade and Migration guide -> Migrate your databases].

Reviewer: ```suggestion
For more details, see link:https://neo4j.com/docs/upgrade-migration-guide/current/version-5/migration/migrate-databases/[Upgrade and Migration guide -> Migrate your databases].
```
